include/mgba-util/common.h

```diff
@@ -46,6 +46,7 @@ typedef intptr_t ssize_t;
 #define snprintf _snprintf
 #define strdup _strdup
 #define lseek _lseek
+#define __builtin_bswap32 _byteswap_ulong
```

Comment:
Please provide a different implementation of `LOAD_32BE` instead of `__builtin_bswap32`.

Reply:
```
#define LOAD_32BE(DEST, ADDR, ARR) { \
	uint32_t _val = ((uint32_t*) ARR)[(ADDR) >> 2]; \
	DEST = (_val << 24) | ((_val & 0xff00) << 8) | ((_val & 0xff0000) >> 8) | (_val >> 24); \
}
```

??

Reply:
No, I meant using _byteswap_ulong instead of __builtin_bswap32 for the MSVC build.

Reply:
```
#ifdef _MSC_VER
#define LOAD_32BE(DEST, ADDR, ARR) DEST = _byteswap_ulong(((uint32_t*) ARR)[(ADDR) >> 2])
#else
#define LOAD_32BE(DEST, ADDR, ARR) DEST = __builtin_bswap32(((uint32_t*) ARR)[(ADDR) >> 2])
#endif
```

If I messed this up too, I'll resubmit a squashed clean pr when it's proper fixed.